src/commands/trigger.cpp

```diff
@@ -0,0 +1,104 @@
+//
+// trigger.cpp
+//
+// Identification: src/commands/trigger.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+
+#include "commands/trigger.h"
+#include "parser/pg_trigger.h"
+
+namespace peloton {
+namespace commands {
+Trigger::Trigger(const peloton::planner::CreatePlan& plan) {
+  trigger_name = plan.GetTriggerName();
+  trigger_funcname = plan.GetTriggerFuncName();
+  trigger_args = plan.GetTriggerArgs();
+  trigger_columns = plan.GetTriggerColumns();
+  trigger_when = plan.GetTriggerWhen();
+  trigger_type = plan.GetTriggerType();
+}
+/*
+ * Add a trigger to the trigger list and update the summary
+ */
+void TriggerList::AddTrigger(Trigger trigger) {
+  triggers.push_back(trigger);
+  UpdateTypeSummary(trigger.GetTriggerType());
+}
+
+/*
+ * Update a type summary when a new type of trigger is added
+ */
+void TriggerList::UpdateTypeSummary(int16_t type) {
+  types_summary[BEFORE_INSERT_ROW] |= TRIGGER_TYPE_MATCHES(
+      type, TRIGGER_TYPE_ROW, TRIGGER_TYPE_BEFORE, TRIGGER_TYPE_INSERT);
+  types_summary[BEFORE_INSERT_STATEMENT] |= TRIGGER_TYPE_MATCHES(
+      type, TRIGGER_TYPE_STATEMENT, TRIGGER_TYPE_BEFORE, TRIGGER_TYPE_INSERT);
+  types_summary[BEFORE_UPDATE_ROW] |= TRIGGER_TYPE_MATCHES(
+      type, TRIGGER_TYPE_ROW, TRIGGER_TYPE_BEFORE, TRIGGER_TYPE_UPDATE);
+  types_summary[BEFORE_UPDATE_STATEMENT] |= TRIGGER_TYPE_MATCHES(
+      type, TRIGGER_TYPE_STATEMENT, TRIGGER_TYPE_BEFORE, TRIGGER_TYPE_UPDATE);
+  types_summary[BEFORE_DELETE_ROW] |= TRIGGER_TYPE_MATCHES(
+      type, TRIGGER_TYPE_ROW, TRIGGER_TYPE_BEFORE, TRIGGER_TYPE_DELETE);
+  types_summary[BEFORE_DELETE_STATEMENT] |= TRIGGER_TYPE_MATCHES(
+      type, TRIGGER_TYPE_STATEMENT, TRIGGER_TYPE_BEFORE, TRIGGER_TYPE_DELETE);
+  types_summary[AFTER_INSERT_ROW] |= TRIGGER_TYPE_MATCHES(
+      type, TRIGGER_TYPE_ROW, TRIGGER_TYPE_AFTER, TRIGGER_TYPE_INSERT);
+  types_summary[AFTER_INSERT_STATEMENT] |= TRIGGER_TYPE_MATCHES(
+      type, TRIGGER_TYPE_STATEMENT, TRIGGER_TYPE_AFTER, TRIGGER_TYPE_INSERT);
+  types_summary[AFTER_UPDATE_ROW] |= TRIGGER_TYPE_MATCHES(
+      type, TRIGGER_TYPE_ROW, TRIGGER_TYPE_AFTER, TRIGGER_TYPE_UPDATE);
+  types_summary[AFTER_UPDATE_STATEMENT] |= TRIGGER_TYPE_MATCHES(
+      type, TRIGGER_TYPE_STATEMENT, TRIGGER_TYPE_AFTER, TRIGGER_TYPE_UPDATE);
+  types_summary[AFTER_DELETE_ROW] |= TRIGGER_TYPE_MATCHES(
+      type, TRIGGER_TYPE_ROW, TRIGGER_TYPE_AFTER, TRIGGER_TYPE_DELETE);
+  types_summary[AFTER_DELETE_STATEMENT] |= TRIGGER_TYPE_MATCHES(
+      type, TRIGGER_TYPE_STATEMENT, TRIGGER_TYPE_AFTER, TRIGGER_TYPE_DELETE);
+}
+
+/**
+ * execute trigger on each row before inserting tuple.
+ */
+storage::Tuple* TriggerList::ExecBRInsertTriggers(storage::Tuple *tuple) {
+  unsigned i;
+  LOG_INFO("enter into ExecBRInsertTriggers");
+
+  //check valid type
+  if (!types_summary[EnumTriggerType::BEFORE_INSERT_ROW]) {
+    return nullptr;
```

Comment:
Thanks. Sure we will prints some logs here. It's acceptable that some tables don't have a specific type of triggers or don't have any triggers, so we won't consider throwing exceptions for now.